amazon/ion/ion_py_objects.py

```diff
@@ -0,0 +1,605 @@
+from decimal import Decimal
+from collections.abc import MutableMapping
+
+from amazon.ion.core import IonType, IonEvent
+from amazon.ion.symbols import SymbolToken
+
+
+class IonPyNull_new(object):
+    ion_annotations = ()
+    ion_type = IonType.NULL  # TODO initialized to NULL type first, what's the real type?
+
+    __name__ = 'IonPyNull_new'
+    __qualname__ = 'IonPyNull_new'
+
+    def __init__(self, ion_type=None, value=None, annotations=()):
+        self.ion_type = ion_type
+        self.ion_annotations = annotations
+
+    def __bool__(self):
+        return False
+
+    def _copy(self):
+        args, kwargs = self._to_constructor_args(self)
+        value = self.__class__(*args, **kwargs)
+        value.ion_type = self.ion_type
+        value.ion_annotations = self.ion_annotations
+        return value
+
+    @staticmethod
+    def _to_constructor_args(value):
+        return (None, value,), {}
+
+    @classmethod
+    def from_event(cls, ion_event):
+        if ion_event.value is not None:
+            args, kwargs = cls._to_constructor_args(ion_event.value)
+        else:
+            args, kwargs = (None, None, ()), {}
+        value = cls(*args, **kwargs)
+        value.ion_type = ion_event.ion_type
+        value.ion_annotations = ion_event.annotations
+        return value
+
+    @classmethod
+    def from_value(cls, ion_type, value, annotations=()):
```

Comment:
Yep, I left it here in case of breaking any existing use case and unit test. But as you mentioned above about the break change, I can remove all of them.

Another interesting thing is that I saw the performance improvement by switch between using and not using this `from_value`. It can improve the performance by ~5-10%.